plugin/ale/cursor.vim

```diff
@@ -81,7 +95,9 @@ function! ale#cursor#EchoCursorWarning(...)
     let index = s:BinarySearch(loclist, pos[1], pos[2])
 
     if index >= 0
-        call ale#cursor#TruncatedEcho(loclist[index]['text'])
+        let l = loclist[index]
+        let msg = s:GetMessage(l.linter_name, l.type, l.text)
```

Comment:
I wonder if it might be easier to pass the whole loclist item to the function instead.